src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -83,48 +73,43 @@ protected ResolveOperationBase(ISharingLifetimeScope mostNestedLifetimeScope, IR
         }
 
         /// <summary>
-        /// Gets the active resolve request.
+        /// Gets or sets the active resolve request.
         /// </summary>
-        public ResolveRequestContextBase? ActiveRequestContext { get; private set; }
+        private ResolveRequestContextBase? ActiveRequestContext { get; set; }
 
         /// <summary>
-        /// Gets the current lifetime scope of the operation; based on the most recently executed request.
+        /// Gets or sets the current lifetime scope of the operation; based on the most recently executed request.
         /// </summary>
-        public ISharingLifetimeScope CurrentScope { get; private set; }
+        private ISharingLifetimeScope CurrentScope { get; set; }
 
         /// <summary>
-        /// Gets the set of all in-progress requests on the request stack.
+        /// Gets the modifiable active request stack.
         /// </summary>
-        public IEnumerable<ResolveRequestContextBase> InProgressRequests => RequestStack;
```

Comment:
If an extension wanted to know the in-progress requests to do anything different with the Circular Dependency Analysis, this is how they get a read-only view of the current request stack.